TicTacToe/PlayField.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include <vector>
+#include <cassert>
+
+using namespace std;
+
+class PlayField {
+public:
+    enum FieldStatus { fsInvalid, fsCrossesWin, fsNoughtsWin, fsDraw, fsNormal };
+    enum CellState { csEmpty, csCross, csNought };
+    class CellIdx {
+    public:
+        static CellIdx CreateIndex(int x, int y) {
+            assert(x >= 0 && y >= 0 && x < DIM && y < DIM);
+            return { x, y };
+        }
+
+        int X() const {
+            return x;
+        }
+
+        int Y() const {
+            return y;
+        }
+    private:
+        int x;
+        int y;
+        CellIdx(int x, int y) {
+            this->x = x;
+            this->y = y;
```

Comment:
В таких случаях this-> писать не надо. Вы в методе, т.е. внутри класса.
```suggestion
            x = iX;
            y = iY;
```
А перекрывающиеся идентификаторы -- это очень нехорошо.